pkg/urbit/vere/pier.c

```diff
@@ -145,312 +137,129 @@ _pier_disk_commit_request(u3_writ* wit_u)
   }
 }
 
-/* _pier_disk_write_header_complete(): commit complete.
-*/
+
 static void
-_pier_disk_write_header_complete(void* vod_p)
+_pier_db_write_header(u3_pier* pir_u,
+                      u3_noun who,
+                      u3_noun is_fake,
+                      u3_noun life)
 {
-  //  no-op, callback required by u3_foil_append()
-  //
+  u3m_lmdb_write_identity(pir_u->log_u->db_u,
+                          who, is_fake, life);
 }
 
-/* _pier_disk_write_header(): save boot metadata.
-*/
+/* _pier_db_read_header(): reads the ships metadata from lmdb
+ */
 static void
-_pier_disk_write_header(u3_pier* pir_u, u3_atom mat)
+_pier_db_read_header(u3_pier* pir_u)
 {
-  //  XX deduplicate with _pier_disk_commit_request
-  //
-  u3_disk* log_u = pir_u->log_u;
-
-  c3_assert( 0ULL == log_u->fol_u->end_d );
-
-  c3_d  len_d = u3r_met(6, mat);
-  c3_d* buf_d = c3_malloc(8 * len_d);
+  u3_noun who, is_fake, life;
+  u3m_lmdb_read_identity(pir_u->log_u->db_u,
+                         &who, &is_fake, &life);
 
-  u3r_chubs(0, len_d, buf_d, mat);
+  _pier_boot_set_ship(pir_u, u3k(who), u3k(is_fake));
+  pir_u->lif_d = u3r_chub(0, life);
 
-  u3_foil_append(_pier_disk_write_header_complete,
-                 (void*)0,
-                 log_u->fol_u,
-                 buf_d,
-                 len_d);
+  u3z(who);
+  u3z(is_fake);
+  u3z(life);
 }
 
-/* _pier_disk_read_header_complete():
-*/
-static void
-_pier_disk_read_header_complete(u3_disk* log_u, u3_noun dat)
+static c3_o
+_pier_db_on_commit_loaded(u3_pier* pir_u,
+                          c3_d id,
+                          u3_noun mat,
+                          u3_noun ovo)
 {
-  u3_pier* pir_u = log_u->pir_u;
-
-  {
-    u3_noun who, fak, len;
-
-    u3x_trel(dat, &who, &fak, &len);
+  u3_noun evt = u3h(u3t(ovo));
+  u3_noun job = u3k(u3t(u3t(u3t(ovo))));
+  c3_d evt_d = u3r_chub(0, evt);
```

Comment:
I think this is actually worth thinking about a bit. My immediate take is that we should be an event log and that persisting the entire `[%work ...]` structure is a historical accident. However, I'm not 100% sure of this and could be convinced that a store of IPC messages is actually the correct thing, though I'm not sure what argument would sway me to that point of view.

Since this PR doesn't change what we're doing, do you mind if I queue up this work for a subsequent PR?

Reply:
I don't mind at all -- I just felt compelled to type it up.